src/scripts/main.js

```diff
@@ -1,3 +1,247 @@
+  createLabel('Office:', officeSelect, form);
+
+  const ageInput = createInput('age', 'number');
+  createLabel('Age:', ageInput, form);
+
+  const salaryInput = createInput('salary', 'number');
+  createLabel('Salary:', salaryInput, form);
+
+  const saveButton = document.createElement('button');
+  saveButton.type = 'submit';
+  saveButton.textContent = 'Save to table';
+
+  form.addEventListener('submit', (event) => {
+    event.preventDefault();
+
+    const newEmployeeData = [...form.elements];
+    const newRow = document.createElement('tr');
+    let index = 0;
+    const newRowData = {};
+
+    for (const element of newEmployeeData) {
+      if (element.tagName === 'BUTTON') {
+        continue;
+      }
+      const cell = document.createElement('td');
+
+      if (element.tagName === 'SELECT') {
+        cell.textContent = officeSelect.selectedOptions[0].innerText;
+      }
+
+      if (element.name === 'salary') {
+        cell.textContent = getSalary(element.value);
+      } else {
+        cell.textContent = element.value;
+      }
+
+      newRowData[tHeadCells[index].innerText.toLowerCase()] = cell.textContent;
+      index++;
+      newRow.append(cell);
+    }
+
+    if (formDataValidation(newRowData)) {
+      tbody.append(newRow)
+      form.reset();
+    }
+  });
+
+  form.append(saveButton);
+  body.append(form);
+}
+
+function createLabel(innerTextForLabel, inputElement, form) {
+  const label = document.createElement('label');
+  label.textContent = innerTextForLabel;
+  label.append(inputElement);
+
+  form.append(label);
+
+  return label;
+}
+
+function createInput(name, type = 'text') {
+  const input = document.createElement('input');
+  input.required = true;
+  input.setAttribute('data-qa', name);
+  input.name = name;
+  input.type = type;
+
+  if (type === 'number') {
+    input.min = '0';
+  }
+
+  return input;
+}
+
+function createSelect(name, options) {
+  const select = document.createElement('select');
+  select.name = name;
+  select.setAttribute('data-qa', name);
+
+  for (const option of options) {
+    const optionHTML = document.createElement('option')
+    optionHTML.value = option;
+    optionHTML.textContent = option;
+    select.append(optionHTML);
+  }
+  return select;
+}
+
+function getSalary(number) {
+  return '$'
+    + parseInt(number).toLocaleString('en-US');
+}
+
+function formDataValidation(newRowData) {
+  let isCorrect = true;
+
+  const notification = document.createElement('div');
+  notification.classList.add('notification');
+  notification.setAttribute('data-qa', 'notification');
+  notification.style.textAlign = 'center';
+
+  const notificationTitle = document.createElement('h2');
+  notificationTitle.classList.add('title');
+
+  const notificationMessage = document.createElement('p');
+
+  const newEmployeeName = newRowData.name;
+  const newEmployeeAge = newRowData.age;
+
+  if (newEmployeeName.length < 4 || parseInt(newEmployeeAge) < 18 || parseInt(newEmployeeAge) > 90) {
```

Comment:
Use parseInt(newEmployeeAge, 10) instead of parseInt(newEmployeeAge) to ensure base 10 conversion.